src/config/parser.rs

```diff
@@ -182,48 +171,60 @@ impl VariantExpr {
             .unwrap_or(false)
         {
             return Err(ParseError::from(ParseErrorType::Custom(
-                "Must have at least one option",
+                "Variant expression have at least one option",
             )));
         }
-        let mut specs = Vec::new();
-        loop {
-            specs.push(Spec::parse(iter)?);
-            if !eat!(iter, Comma) {
-                break;
+        Ok(VariantExpr {
+            specs: CommaList::parse(iter, TokType::RBracket)?.list,
+        })
+    }
+}
+
+// match-expr -> { comma-list<(expr ":" spec)> }
+impl SimpleParse for MatchExpr {
+    fn parse<I: Iterator<Item = Token>>(iter: &mut Peekable<I>) -> ParseResult<Self> {
+        // Allow `expr ":" spec` to be parsed into a tuple `(expr, spec)`.
+        impl SimpleParse for (Expr, Spec) {
+            fn parse<I: Iterator<Item = Token>>(iter: &mut Peekable<I>) -> ParseResult<Self> {
+                let expr = Expr::parse(iter)?;
+                expect(iter, &[TokType::Colon])?;
+                let spec = Spec::parse(iter)?;
+                Ok((expr, spec))
             }
         }
-        expect(iter, &[TokType::RBracket])?;
-        Ok(VariantExpr { specs })
+        expect(iter, &[TokType::LBrace])?;
+        Ok(MatchExpr {
+            cases: CommaList::parse(iter, TokType::RBrace)?.list,
+        })
     }
 }
 
-// match-expr -> { (expr ":" spec ":")* "default" ":" spec }
-impl MatchExpr {
-    pub fn parse<I: Iterator<Item = Token>>(iter: &mut Peekable<I>) -> ParseResult<MatchExpr> {
-        expect(iter, &[TokType::LBrace])?;
-        let mut cases = Vec::new();
+// comma-list<T> -> (T ",")* T?
+impl<T: SimpleParse> CommaList<T> {
+    pub fn parse<I: Iterator<Item = Token>>(
+        iter: &mut Peekable<I>,
+        end: TokType,
+    ) -> ParseResult<Self> {
+        let mut list = Vec::new();
         loop {
-            // Allow trailing commas
-            if eat!(iter, RBrace) {
+            // Allow trailing comma
```

Comment:
Explain how this permits a trailing comma. 

There is also a repetitive if statement in this loop:

```rust
            if eat(iter, end) {
                break;
            }
```
This should be refactored if possible.